### PR TITLE
Added handicap, should fix #32

### LINKED ISSUE #32: Add a handicap feature (q3a-style)
The title says it all. Would be cool to have a handicap feature to allow people to play against others who are on a different skill level. I can't play some of my friends because it makes no sense dut to our differences in skill (match results are usually something like 30:0).

How handicap works: * each player has a handicap value that he can set in his config. It defaults to 100. * the weaker player does nothing (so he has 100) * the stronger player specifies a lower handicap value (like 70). This affects (weakens) him in several ways:

    he starts with <handycap-value> health instead of 100
    he can't pick up more than <handycap-value> health with normal medkits (but more with +5 bubbles and MH)
    if he hurts another player, he only does <handycap-value>% (per cent) of the normal damage 

There's no need to implement all of these (and others may also be introduced, of course), but a feature that does something similar to the things described here would allow me to attract a few of my friends to q2w. :)

Changed 14 months ago by jdolan

    owner changed from jdolan to wickedshell
    status changed from new to assigned

A userinfo cvar on the client could set this for the server, and the game module could then honor it in G_ClientRespawn_.

### PR DESCRIPTION
**Pull request by kaadmy**
As the title says, this adds handicap in the form of reducing reducing max health and max overheal health.

### CODE REVIEW

File: src/client/cl_main.c [hunk 1]
```diff
@@ -44,6 +44,7 @@ cvar_t *name;
 cvar_t *password;
 cvar_t *rate;
 cvar_t *skin;
+cvar_t *handicap;
```
Comment (jdolan) [inline src/client/cl_main.c]: Should be a corresponding extern declaration of this in cl_main.h.

File: src/game/default/g_types.h [hunk 2]
```diff
@@ -698,6 +698,8 @@ typedef struct {
 	char skin[MAX_QPATH];
 	g_hand_t hand;
 
+	vec_t handicap; // handicap inverse percentage from 0 to 100
+
```
Comment (jdolan) [inline src/game/default/g_types.h]: Will this eventually influence G_Damage?

Comment (kaadmy) [inline src/game/default/g_types.h]: I don't know, I should be able to add it easily.

Comment (jdolan) [inline src/game/default/g_types.h]: I never used Q3's handicap feature, so I'm not familiar with all of it's functionality. But from what @maci0 said above, apparently your handicap would also make your weapons that much less effective. I do not have a strong opinion on whether that is needed or not.

Comment (kaadmy) [inline src/game/default/g_types.h]: I'll add it to allow RJing still :)

File: src/client/cl_main.c (hunk 1 above)
Comment (kaadmy) [inline src/client/cl_main.c]: Okay, it's fixed

**Comment by kaadmy** [comment]
Oh, I was still changing this branch :(

**Comment by jdolan** [comment]
Restored.

**Pull request merged**